controller/linodecluster_controller_test.go

```diff
@@ -0,0 +1,357 @@
+// /*
+// Copyright 2023 Akamai Technologies, Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// */
+
+package controller
+
+import (
+	"go.uber.org/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	infrav1 "github.com/linode/cluster-api-provider-linode/api/v1alpha1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("lifecycle", Ordered, Label("cluster", "lifecycle"), func() {
+	var mockCtrl *gomock.Controller
```

Comment:
It doesn't appear to be using a mock client here. If no mock client is being configured, the test is making real calls to the Linode API.

Reply:
I've removed all mentions of mock client now. We are not using the client because we are only creating CAPL and CAPI clusters and no calls to the Linode API at all.